### PR TITLE
Remove Label

### PR DESCRIPTION
**Pull request by Hirobreak**
- Removing label, sending peer event
- Parsing delete label command 320

### CODE REVIEW

File: electron_app/src/ipc/database.js [hunk 1]
```diff
@@ -149,6 +149,8 @@ ipc.answerRenderer('db-get-labesls-by-text', text =>
   dbManager.getLabelsByText(text)
 );
 
+ipc.answerRenderer('db-get-labeluuid', dbManager.getLabelByUuid);
```
Comment (erikaperugachi) [inline electron_app/src/ipc/database.js]: rename: `db-get-label-by-uuid`

File: email_mailbox/src/actions/labels.js [hunk 2]
```diff
@@ -48,13 +48,26 @@ export const addLabels = labels => {
...
+  };
+};
+
+export const removeLabel = (id, uuid) => {
   return async dispatch => {
     try {
       const response = await deleteLabelById(id);
-      if (response) {
-        dispatch(removeLabelOnSuccess(id));
+      if (!response) {
```
Comment (erikaperugachi) [inline email_mailbox/src/actions/labels.js]: `if (!response) return` .... delete brakets just by one line

File: email_mailbox/src/actions/labels.js [hunk 3]
```diff
@@ -48,13 +48,26 @@ export const addLabels = labels => {
   };
 };
 
-export const removeLabel = id => {
+export const removeLabels = labelIds => {
+  return {
+    type: Label.REMOVE_BATCH,
```
Comment (erikaperugachi) [inline email_mailbox/src/actions/labels.js]: rename `REMOVE_LABELS`

File: email_mailbox/src/components/PanelWrapper.js [hunk 4]
```diff
@@ -398,8 +409,8 @@ class PanelWrapper extends Component {
         this.props.onUpdateUnreadEmailsBadge(labelIdsBadge);
     }
 
-    if (activity || label) {
-      this.props.onAddDataApp({ activity, label });
+    if (activity || label || deletedLabels) {
+      this.props.onAddDataApp({ activity, label, deletedLabels });
```
Comment (erikaperugachi) [inline email_mailbox/src/components/PanelWrapper.js]: wrong: this cross... label and deleteLabels are from the same reducer, jusy one action should happen.

File: email_mailbox/src/components/SettingLabels.js [hunk 5]
```diff
@@ -104,14 +104,14 @@ const renderCustomLabelItem = (index, customLabelItem, props) => (
         status={customLabelItem.visible ? 'all' : 'none'}
       />
     </div>
-    {props.exist && (
```
Comment (erikaperugachi) [inline email_mailbox/src/components/SettingLabels.js]: Delete `exist` from `PropTypes`

File: email_mailbox/src/reducers/__tests__/labels.js [hunk 6]
```diff
@@ -118,4 +118,12 @@ describe('Label actions - UPDATE', () => {
     const newState = labelReducer(state, action);
     expect(newState).toBe(new Map({}));
   });
+
+  it('should remove group of label', () => {
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/__tests__/labels.js]: rename `should remove labels by labelIds`

File: email_mailbox/src/reducers/__tests__/labels.js [hunk 7]
```diff
@@ -118,4 +118,12 @@ describe('Label actions - UPDATE', () => {
...
+    const state = initState(myLabels);
+    const labelIds = ['1', '6'];
+    const action = actions.removeLabels(labelIds);
+    const newState = labelReducer(state, action);
+    expect(newState.size).toBe(5);
+  });
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/__tests__/labels.js]: Add test:
- should not remove labels nonexisten ... (check the result)
- should not remove labels by wrong emailIds ... (use labelsIds as number array: [1,6])

File: email_mailbox/src/reducers/crossReducer.js [hunk 8]
```diff
@@ -37,7 +43,7 @@ export const crossReducer = (state = new Map(), action) => {
         feeditems: feeditem
           ? feeditems(feeditemsState, feeditem)
           : feeditemsState,
-        labels: label ? labels(labelsState, label) : labelsState,
+        labels: deletedLabelsState,
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/crossReducer.js]: delete

File: email_mailbox/src/reducers/crossReducer.js [hunk 9]
```diff
@@ -37,7 +43,7 @@ export const crossReducer = (state = new Map(), action) => {
         feeditems: feeditem
           ? feeditems(feeditemsState, feeditem)
           : feeditemsState,
-        labels: label ? labels(labelsState, label) : labelsState,
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/crossReducer.js]: retrieve

File: email_mailbox/src/reducers/crossReducer.js [hunk 10]
```diff
@@ -27,6 +28,11 @@ export const crossReducer = (state = new Map(), action) => {
       const feeditemsState = state.get('feeditems');
       const labelsState = state.get('labels');
       const threadsState = state.get('threads');
+
+      const addedLabelsState = label ? labels(labelsState, label) : labelsState;
+      const deletedLabelsState = deletedLabels
+        ? labels(addedLabelsState, deletedLabels)
+        : addedLabelsState;
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/crossReducer.js]: delete

File: email_mailbox/src/reducers/crossReducer.js [hunk 11]
```diff
@@ -27,6 +28,11 @@ export const crossReducer = (state = new Map(), action) => {
       const feeditemsState = state.get('feeditems');
       const labelsState = state.get('labels');
       const threadsState = state.get('threads');
+
+      const addedLabelsState = label ? labels(labelsState, label) : labelsState;
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/crossReducer.js]: delete

File: email_mailbox/src/reducers/crossReducer.js [hunk 12]
```diff
@@ -27,6 +28,11 @@ export const crossReducer = (state = new Map(), action) => {
...
+      const deletedLabelsState = deletedLabels
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/crossReducer.js]: delete

File: email_mailbox/src/reducers/crossReducer.js [hunk 13]
```diff
@@ -27,6 +28,11 @@ export const crossReducer = (state = new Map(), action) => {
...
+        ? labels(addedLabelsState, deletedLabels)
```
Comment (erikaperugachi) [inline email_mailbox/src/reducers/crossReducer.js]: delete

File: email_mailbox/src/utils/electronEventInterface.js [hunk 14]
```diff
@@ -942,6 +954,19 @@ const handlePeerLabelCreated = async ({ rowid, params }) => {
   return { rowid };
 };
 
+const handlePeerLabelDelete = async ({ rowid, params }) => {
+  const { uuid } = params;
+  const [label] = await getLabelByUuid(uuid);
+  if (!label) {
+    return { rowid };
```
Comment (erikaperugachi) [inline email_mailbox/src/utils/electronEventInterface.js]: just one line

File: email_mailbox/src/utils/electronEventInterface.js [hunk 15]
```diff
@@ -942,6 +954,19 @@ const handlePeerLabelCreated = async ({ rowid, params }) => {
...
+  }
+  const response = await deleteLabelById(label.id);
+  if (!response) {
+    return { rowid: null };
```
Comment (erikaperugachi) [inline email_mailbox/src/utils/electronEventInterface.js]: just one line

File: email_mailbox/src/components/PanelWrapper.js (hunk 4 above)
Comment (Hirobreak) [reply email_mailbox/src/components/PanelWrapper.js]: Should I add something like an ADD_DELETE_LABELS action instead?

File: email_mailbox/src/reducers/__tests__/labels.js (hunk 7 above)
Comment (Hirobreak) [reply email_mailbox/src/reducers/__tests__/labels.js]: Could you explain this one a bit more `should not remove labels by wrong emailIds`?

**Pull request merged**